### PR TITLE
Add mixin for the IBM-DB2 integration

### PR DESCRIPTION
**Pull request by dpaasman00**
This mixin adds 1 dashboard and 3 alerts

Screenshot 2023-06-19 at 2 28 49 PM
Screenshot 2023-06-19 at 2 28 58 PM

### CODE REVIEW

File: mixin/.lint [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    reason: "Prometheus datasource variable is being named as prometheus_datasource now while linter expects 'datasource'"
+  panel-datasource-rule:
+    reason: "Loki datasource variable is being named as loki_datasource now while linter expects 'datasource'"
+  template-datasource-rule:
+    reason: "Based on new convention we are using variable names prometheus_datasource and loki_datasource where as linter expects 'datasource'"
+  template-instance-rule:
+    reason: "Based on new convention we are using variable names prometheus_datasource and loki_datasource where as linter expects 'datasource'"
+  panel-units-rule:
+    reason: "Custom units are used for better user experience in these panels"
+    entries:
```
Comment (schmikei) [inline mixin/.lint:12]: Is this unfinished?

File: mixin/jsonnetfile.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+          "git": {
+            "remote": "https://github.com/grafana/grafonnet-lib.git",
+            "subdir": "grafonnet"
+          }
+        },
+        "version": "master"
+      }
+    ],
+    "legacyImports": true
+  }
```
Comment (schmikei) [inline mixin/jsonnetfile.json:15]:
```suggestion
  }
  
```

File: mixin/jsonnetfile.lock.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+            "remote": "https://github.com/grafana/grafonnet-lib.git",
+            "subdir": "grafonnet"
+          }
+        },
+        "version": "30280196507e0fe6fa978a3e0eaca3a62844f817",
+        "sum": "342u++/7rViR/zj2jeJOjshzglkZ1SY+hFNuyCBFMdc="
+      }
+    ],
+    "legacyImports": false
+  }
```
Comment (schmikei) [inline mixin/jsonnetfile.lock.json:16]:
```suggestion
  }
  
```

File: mixin/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+The IBM DB2 mixin consists of a configurable Grafana dashboard and alerts based on the [IBM DB2 exporter](../README.md).
+
+The IBM DB2 mixin contains the following dashboards:
+- IBM DB2 overview
+
+## IBM DB2 overview
+The IBM DB2 overview dashboard provides details about the general state of the database like bufferpool hit ratio, active connections, and deadlocks.
+
+![First screenshot of the overview dashboard](https://storage.googleapis.com/grafanalabs-integration-assets/snowflake/screenshots/ibm_db2_overview_1.png)
```
Comment (schmikei) [inline mixin/README.md:11]: I would leave a TODO to get screenshots uploaded since these 404

File: mixin/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# IBM DB2 Mixin
+
+The IBM DB2 mixin consists of a configurable Grafana dashboard and alerts based on the [IBM DB2 exporter](../README.md).
+
+The IBM DB2 mixin contains the following dashboards:
+- IBM DB2 overview
```
Comment (schmikei) [inline mixin/README.md:6]:
```suggestion

- IBM DB2 overview
```

Just adding whitespace :)

**Review by schmikei — CHANGES_REQUESTED** [review]
Just some small changes. Overall looks good and passes the lint :)

File: mixin/.lint [hunk 6]
```diff
@@ -0,0 +1,9 @@
+exclusions:
+  template-job-rule:
+    reason: "Prometheus datasource variable is being named as prometheus_datasource now while linter expects 'datasource'"
+  panel-datasource-rule:
+    reason: "Loki datasource variable is being named as loki_datasource now while linter expects 'datasource'"
+  template-datasource-rule:
+    reason: "Based on new convention we are using variable names prometheus_datasource and loki_datasource where as linter expects 'datasource'"
+  template-instance-rule:
+    reason: "Based on new convention we are using variable names prometheus_datasource and loki_datasource where as linter expects 'datasource'"
```
Comment (kuiperda) [inline mixin/.lint:9]: Would actually add this last (empty) rule back in, as lint fails without it but specific panel entries aren't actually needed. 

run `make lint` to see this

File: mixin/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# IBM DB2 Mixin
+
+The IBM DB2 mixin consists of a configurable Grafana dashboard and alerts based on the [IBM DB2 exporter](../README.md).
+
+The IBM DB2 mixin contains the following dashboards:
+
+- IBM DB2 overview
+
```
Comment (kuiperda) [inline mixin/README.md:8]: Usually we've added a few other things to the readme, namely alerts and how to configure disabling Loki logs. go to jsonnet-libs and checkout feat/sap-hana for an example readme.

Comment (kuiperda) [reply mixin/README.md:8]: pretty much just these sections
```
and the following alerts:

- SapHanaHighCpuUtilization
- SapHanaHighPhysicalMemoryUsage
- SapHanaMemAllocLimitBelowRecommendation
- SapHanaHighMemoryUsage
- SapHanaHighDiskUtilization
- SapHanaHighSqlExecutionTime
- SapHanaHighReplicationShippingTime
- SapHanaReplicationStatusError

Default thresholds can be configured in `config.libsonnet`.

```js
{
  _config+:: {
    alertsCriticalHighCpuUsage: 80, // percent 0-100
    alertsCriticalHighPhysicalMemoryUsage: 80, // percent 0-100
    alertsWarningLowMemAllocLimit: 90, // percent 0-100
    alertsCriticalHighMemoryUsage: 80, // percent 0-100
    alertsCriticalHighDiskUtilization: 80, //percent 0-100
    alertsCriticalHighSqlExecutionTime: 1, // second
    alertsCriticalHighReplicationShippingTime: 1, //second
  },
}
```

```
To get SAP HANA trace logs, [Promtail and Loki needs to be installed](https://grafana.com/docs/loki/latest/installation/) and provisioned for logs with your Grafana instance. The default directory for trace files is `/opt/hana/shared/<sid>/HDB<insnr>/<host>/trace/` e.g. `/opt/hana/shared/ID0/HDB00/hana-0/trace/*.trc`

SAP HANA trace logs are enabled by default in the `config.libsonnet` and can be removed by setting `enableLokiLogs` to `false`. Then run `make` again to regenerate the dashboard:


{
  _config+:: {
    enableLokiLogs: false,
  },
}


## Alerts overview

SapHanaHighCpuUtilization: CPU utilization is high.
SapHanaHighPhysicalMemoryUsage: Current physical memory usage of the host is approaching capacity.
SapHanaMemAllocLimitBelowRecommendation: Memory allocation limit set below recommended limit.
SapHanaHighMemoryUsage: Current SAP HANA memory usage is approaching capacity.
SapHanaHighDiskUtilization: SAP HANA disk is approaching capacity.
SapHanaHighSqlExecutionTime: SAP HANA SQL average execution time is high.
SapHanaHighReplicationShippingTime: SAP HANA system replication log shipping delay is high.
SapHanaReplicationStatusError: SAP HANA system replication status signifies an error.
```

File: mixin/config.libsonnet [hunk 8]
```diff
@@ -0,0 +1,15 @@
+{
+  _config+:: {
+    dashboardTags: ['ibm-db2-mixin'],
+    dashboardPeriod: 'now-3h',
+    dashboardTimezone: 'default',
+    dashboardRefresh: '1m',
+
+    // alerts thresholds
+    alertsHighLockWaitTime: 2000,
```
Comment (kuiperda) [inline mixin/config.libsonnet:9]: update this with the comments for units

File: mixin/jsonnetfile.lock.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    "version": 1,
+    "dependencies": [
+      {
+        "source": {
+          "git": {
+            "remote": "https://github.com/grafana/grafonnet-lib.git",
+            "subdir": "grafonnet"
+          }
+        },
+        "version": "30280196507e0fe6fa978a3e0eaca3a62844f817",
```
Comment (kuiperda) [inline mixin/jsonnetfile.lock.json:11]: delete this lock file and the vendor folder, then run `jb install` in mixin, version can be updated

**Review by kuiperda — CHANGES_REQUESTED** [review]

**Review by kuiperda — APPROVED** [review]

**Pull request closed without merging**